src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -1438,12 +1440,10 @@ protected SCMRevision retrieve(SCMHead head, TaskListener listener) throws IOExc
                     PullRequestSCMHead prhead = (PullRequestSCMHead) head;
 
                     GHPullRequest pr = getDetailedGHPullRequest(prhead.getNumber(), listener, github, ghRepository);
-                    assert(pr.getMergeable() != null);
-                    if (Boolean.FALSE.equals(pr.getMergeable())) {
-                        throw new AbortException("Pull request " + prhead.getNumber() + " is not mergeable.");
-                    }
                     PullRequestSCMRevision prRev = createPullRequestSCMRevision(pr, prhead, listener, github, ghRepository);
-                    prRev.validateMergeHash(ghRepository);
+                    if (prhead.isMerge()) {
```

Comment:
Same as above.  Check moved to validation method and made validation only run in merge case.